cmake/sdlchecks.cmake

```diff
@@ -82,29 +82,33 @@ endmacro()
 # - SDL_ALSA_SHARED opt
 # - HAVE_SDL_LOADSO opt
 macro(CheckALSA)
-  if(SDL_ALSA)
-    CHECK_INCLUDE_FILE(alsa/asoundlib.h HAVE_ASOUNDLIB_H)
-    if(HAVE_ASOUNDLIB_H)
-      CHECK_LIBRARY_EXISTS(asound snd_pcm_recover "" HAVE_LIBASOUND)
-    endif()
-    if(HAVE_LIBASOUND)
-      set(HAVE_ALSA TRUE)
-      file(GLOB ALSA_SOURCES ${SDL2_SOURCE_DIR}/src/audio/alsa/*.c)
-      list(APPEND SOURCE_FILES ${ALSA_SOURCES})
-      set(SDL_AUDIO_DRIVER_ALSA 1)
-      if(SDL_ALSA_SHARED AND NOT HAVE_SDL_LOADSO)
-        message_warn("You must have SDL_LoadObject() support for dynamic ALSA loading")
-      endif()
-      FindLibraryAndSONAME("asound")
-      if(SDL_ALSA_SHARED AND ASOUND_LIB AND HAVE_SDL_LOADSO)
-        set(SDL_AUDIO_DRIVER_ALSA_DYNAMIC "\"${ASOUND_LIB_SONAME}\"")
-        set(HAVE_ALSA_SHARED TRUE)
-      else()
-        list(APPEND EXTRA_LIBS asound)
-      endif()
-      set(HAVE_SDL_AUDIO TRUE)
+    if(SDL_ALSA)
+        find_package(ALSA MODULE)
+        if(ALSA_FOUND)
+            list(APPEND EXTRA_INCLUDE_DIR ${ALSA_INCLUDE_DIRS})
+            list(APPEND EXTRA_LIBS ${ALSA_LIBRARIES})
+            file(GLOB ALSA_SOURCES "${SDL2_SOURCE_DIR}/src/audio/alsa/*.c")
+            list(APPEND SOURCE_FILES ${ALSA_SOURCES})
+            set(SDL_AUDIO_DRIVER_ALSA 1)
+            set(HAVE_ALSA TRUE)
+            if(SDL_ALSA_SHARED)
+                if(HAVE_SDL_LOADSO)
+                    FindLibraryAndSONAME("asound")
```

Comment:
Does this succeed with vcpkg, when building with `-DSDL_ALSA_SHARED=ON`?
I mean, no `Unable to find asound shared object` warning is printed?